plugins/ua_network_tcp.c

```diff
@@ -1168,12 +1168,8 @@ UA_ClientConnectionTCP(UA_ConnectionConfig conf,
             connected = true;
             break;
         }
-        /*fix double free (clang)*/
```

Comment:
If these lines were replaced by `ClientNetworkLayerTCP_close` clang compiler will complain. This is because in `ClientNetworkLayerTCP_close` there is this condition check `if(tcpConnection->server)` before we really free `tcpConnection->server`. However I think it's only a problem of the compiler...

Reply:
Ok. Thank you for your explanation. But we need to fix it without changing UA_ClientConnectionTCP

clang is often right.

Maybe setting tcpConnection->server and connection->handle at NULL after the free can resolve the issue.